cpu/samd5x/periph/timer.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (C) 2018 Mesotic SAS
+ *
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_saml1x
+ * @ingroup     drivers_periph_timer
+ * @{
+ *
+ * @file        timer.c
+ * @brief       Low-level timer driver implementation
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ *
+ * @}
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "board.h"
+#include "cpu.h"
+
+#include "periph/timer.h"
+#include "periph_conf.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/**
+ * @brief Timer state memory
+ */
+static timer_isr_ctx_t config[TIMER_NUMOF];
+
+/* enable timer interrupts */
+static inline void _irq_enable(tim_t dev);
+
+
+/**
+ * @brief Setup the given timer
+ */
+int timer_init(tim_t dev, unsigned long freq, timer_cb_t cb, void *arg)
+{
+    /* at the moment, the timer can only run at 3MHz */
+    if (freq != 32768ul) {
```

Comment:
Comment says "3MHz", but this number is 32768. Is that correct?
BTW. The value of `freq` is unused (same in saml1x).

Reply:
The code duplication will be resolved by #11336.